### PR TITLE
[NU-1303] ComponentId -> DesignerWideComponentId, ComponentInfo -> ComponentId

### PR DESCRIPTION
**Pull request by arkadius**
## Describe your changes
Changes:
- `ComponentId` -> `DesignerWideComponentId`
- `ComponentInfo` -> `ComponentId`
- Added `name` into `ComponentDefinitionWithImplementation` -> removed all places where were tuples (Name, `ComponentDefinitionWithImplementation`),  `(ComponentId, ComponentDefinitionWithImplementation)`, `Map[ComponentId, ComponentDefinitionWithImplementation]` replaced by `List[ComponentDefinitionWithImplementation]` 

## Checklist before merge
- [ ] Related issue ID is placed at the beginning of PR title in \[brackets\] (can be GH issue or Nu Jira issue)
- [ ] Code is cleaned from temporary changes and commented out lines
- [ ] Parts of the code that are not easy to understand are documented in the code
- [ ] Changes are covered by automated tests
- [ ] Showcase in dev-application.conf added to demonstrate the feature
- [ ] Documentation added or updated
- [ ] Added entry in _Changelog.md_ describing the change from the perspective of a public distribution user
- [ ] Added _MigrationGuide.md_ entry in the appropriate subcategory if introducing a breaking change
- [ ] Verify that PR will be squashed during merge

### CODE REVIEW

**Review by lukasz-bigorajski — APPROVED** [review]

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementResources.scala [hunk 1]
```diff
@@ -69,7 +69,7 @@ object ManagementResources {
         "variables" -> a.variables.asJson
       )
 
-    implicit val componentInfoEncoder: Encoder[ComponentInfo]         = deriveConfiguredEncoder
+    implicit val componentInfoEncoder: Encoder[ComponentId]           = deriveConfiguredEncoder
```
Comment (wrzontek) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementResources.scala:72]:
```suggestion
    implicit val componentIdEncoder: Encoder[ComponentId]           = deriveConfiguredEncoder
```

File: designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/definition/package.scala [hunk 2]
```diff
@@ -78,44 +78,41 @@ package object definition {
...
+    implicit val componentIdEncoder: Encoder[ComponentId] = Encoder.encodeString.contramap(_.toString)
 
     implicit val encoder: Encoder[UINodeEdges] = deriveConfiguredEncoder
   }
 
   object UIComponentNodeTemplate {
 
     def create(
-        componentInfo: ComponentInfo,
+        componentId: ComponentId,
```
Comment (wrzontek) [inline designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/definition/package.scala:96]: doesn't this change warrant an entry in the changelog?

Comment (wrzontek) [reply designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/definition/package.scala:96]: (I meant to point to the field in the case class, not the `create` method)

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementResources.scala (hunk 1 above)
Comment (arkadius) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/ManagementResources.scala]: Fixed, in fact this encoder wasn't needed here :)

File: designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/definition/package.scala (hunk 2 above)
Comment (arkadius) [reply designer/restmodel/src/main/scala/pl/touk/nussknacker/restmodel/definition/package.scala:96]: You mean in the migration guide? I've added it now. I don't think that the end-user is interested in such changes, so it shouldn't be in the changelog.

**Review by wrzontek — APPROVED** [review]

**Pull request merged**